chore: update swift package tools [WPB-6090]

markusfassbender opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6090" title="WPB-6090" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-6090</a>  [iOS] Migrate to Xcode 15.2
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Document how to update the swift package tools
- Update [Sourcery](https://github.com/krzysztofzablocki/Sourcery)
  - Add missing run phase to `WireDataModel`
- Update [SwiftLint](https://github.com/realm/SwiftLint)
  - Fix `redundant_void_return` rules, see [release notes](https://github.com/realm/SwiftLint/releases/tag/0.54.0).

### Testing

#### How to Test

No testing required.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

markusfassbender committed:
update sourcery

markusfassbender committed:
add description how to

markusfassbender committed:
update swiftlint

markusfassbender committed:
update generated files

markusfassbender committed:
add missing sourcery for wiredatamodel

markusfassbender committed:
Revert "add missing sourcery for wiredatamodel"

This reverts commit 4a73672bc174d90d5e45576a615fce9c80ecb035.

agisilaos reviewed (commented):
@markusfassbender Please bump the SwiftLint version in .swiftlint.yml file too.

markusfassbender committed:
update swiftlint version

markusfassbender committed:
Merge remote-tracking branch 'origin/develop' into chore/update-swift-package-tools

markusfassbender commented on scripts/run-sourcery.sh:
```diff
@@ -21,7 +21,7 @@ set -Eeuo pipefail
 REPO_ROOT=$(git rev-parse --show-toplevel)
 SCRIPTS_DIR="$REPO_ROOT/scripts"
-SOURCERY="$SCRIPTS_DIR/.build/artifacts/scripts/sourcery/sourcery.artifactbundle/sourcery/bin/sourcery/bin/sourcery"
+SOURCERY="$SCRIPTS_DIR/.build/artifacts/scripts/sourcery/sourcery/bin/sourcery"
```
@caldrian helped me this this, do you know?

caldrian replied:
The structure in the zip file changed upstream. I hope they'll keep the structure from now on.

markusfassbender committed:
bump ci swiftlint

markusfassbender committed:
Merge remote-tracking branch 'origin/develop' into chore/update-swift-package-tools

markusfassbender committed:
update swiftlint path with version

markusfassbender committed:
fix swiftlint redundant_void_return

From release notes:
Rewrite redundant_void_return rule using SwiftSyntax.
Also include redundant void return clauses for closures in addition to
functions. This can be disabled by configuring the rule with include_closures: false.

markusfassbender committed:
Merge remote-tracking branch 'origin/develop' into chore/update-swift-package-tools

caldrian commented on wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Participants.swift:
```diff
@@ -555,7 +555,7 @@ final class ConversationParticipantsTests: ZMConversationTestsBase {
 
     func testThatItRefetchesRolesIfNoRoles() {
 
-        syncMOC.performGroupedAndWait { _ -> Void in
```
Nice, however, maybe these changes should have gone into a separate PR.

markusfassbender replied:
The rule got rewritten with version 0.54.0 and thus lead to failure. Either I could change the swiftlint configuration or just solve the cases...
